buffer/src/main/java/io/netty/buffer/UnpooledHeapByteBuf.java

```diff
@@ -194,7 +194,7 @@ public ByteBuf getBytes(int index, byte[] dst, int dstIndex, int length) {
 
     @Override
     public ByteBuf getBytes(int index, ByteBuffer dst) {
-        checkIndex(index, dst.remaining());
+        ensureAccessible();
```

Comment:
@njhill IHMO the nice thing about `checkIndex` is that the exception message is lot more useful

Reply:
@normanmaurer again the motivation is consistency. All the equivalent places in the unpooled ByteBuf impls avoid the duplicate check when the subsequent operation will throw appropriate `IndexOutOfBoundsException`s. I'd assume if we want the additional message in this very specific method then we should add the (duplicate) checks in all the other places too?

Reply:
I see..